### PR TITLE
HPD Complaints Script

### PR DESCRIPTION
**Pull request by chrishenry**
# Changelog

### New

* HPD Complaints script. [Chris Henry]

### Changes

* bring volumes in line with data-imports folder structure. [Chris Henry]

### Fix

* Send correct # of params to process_df. [Chris Henry]

### CODE REVIEW

File: data-imports/import-hpd-complaints.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+    cmp_description = "HPD Complaints"
+    cmp_input_csv_url = hpd_complaints_csv
+    cmp_sep_char = ","
+    cmp_pickle = os.path.join(hpd_complaints_dir, 'df_complaints.pkl')
+    cmp_table_name = 'hpd_complaints'
+    cmp_load_pickle = args.LOAD_PICKLE
+    cmp_save_pickle = args.SAVE_PICKLE
+    cmp_db_action = args.DB_ACTION
+    cmp_chunk_size = 5000
```
Comment (rdlester) [inline data-imports/import-hpd-complaints.py:93]: Hmm, we've been using 2500 elsewhere, is it better to use larger chunks? Is there something about the HPD complaints that allow for a larger chunk size than the other scripts?

File: data-imports/import-hpd-complaints.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+                cmp_dtype_dict,
+                cmp_load_pickle,
+                cmp_save_pickle,
+                cmp_pickle,
+                cmp_db_action,
+                cmp_truncate_columns,
+                cmp_date_time_columns,
+                cmp_chunk_size,
+                cmp_df_keep_cols
+               )
```
Comment (rdlester) [inline data-imports/import-hpd-complaints.py:109]: Not for this PR but I'd like to drop a couple of these params and just pass the args struct directly, much less copy-pasta.

File: data-imports/utils.py [hunk 3]
```diff
@@ -152,7 +152,7 @@ def hpd_csv2sql(description, input_csv_url, sep_char,
             log.info("Processing chunk {}".format(chunk_num))
 
             chunk_pickle_file = pickle_file.format(chunk_num)
-            df = process_df(df, log, description, save_pickle, chunk_pickle_file,
+            df = process_df(df, log, description, save_pickle, pickle_file,
```
Comment (rdlester) [inline data-imports/utils.py:155]: This is supposed to be chunk_pickle_file - when importing the CSV in chunks (only used for 311 right now), the user provides a pickle file name format, which gets formatted with the number (into chunk_pickle_file). Passing pickle_file will cause each chunk's pickle to overwrite that of the previous chunk.

I should probably pull this stuff out to a separate method so it's less confusing.

**Comment by jpfreeley** [comment]
I'm not sure if there's anything significantly better or worse .. it's just
small amount of overhead .. less dense tables can support higher chunks. I
had basically started everything with 5K and if it didn't work, tried 2.5K
.

File: data-imports/import-hpd-complaints.py (hunk 1 above)
Comment (chrishenry) [reply data-imports/import-hpd-complaints.py:93]: Ahh, this was arbitrary, but it seemed to perform pretty well. `hpd_complaints` is a much narrower table than most of the others, which does allow for larger chunks.

File: data-imports/import-hpd-complaints.py (hunk 2 above)
Comment (chrishenry) [reply data-imports/import-hpd-complaints.py:109]: Yea, that'd be a great improvement. In the spirit of tracking changes we want to make, I'll move this to an issue.

https://github.com/chrishenry/deltanyc-heatseek/issues/33

File: data-imports/utils.py (hunk 3 above)
Comment (chrishenry) [reply data-imports/utils.py:155]: Hmm, I wonder if there was a bad merge somewhere? Or if that param order is in a different PR? The changes I made here seem to correspond to what's in master now for `process_df`.

https://github.com/chrishenry/deltanyc-heatseek/blob/master/data-imports/utils.py#L192

**Pull request merged**